pkg/kubelet/kubelet.go

```diff
@@ -2313,6 +2376,99 @@ func (kl *Kubelet) HandlePodSyncs(pods []*v1.Pod) {
 	}
 }
 
+func (kl *Kubelet) canResizePod(pod *v1.Pod) (bool, *v1.Pod, v1.PodResizeStatus) {
+	var otherActivePods []*v1.Pod
+
+	node, err := kl.getNodeAnyWay()
+	if err != nil {
+		klog.ErrorS(err, "getNodeAnyway function failed")
+		return false, nil, ""
+	}
+	cpuAvailable := node.Status.Allocatable.Cpu().MilliValue()
+	memAvailable := node.Status.Allocatable.Memory().Value()
+	cpuRequests := resource.GetResourceRequest(pod, v1.ResourceCPU)
+	memRequests := resource.GetResourceRequest(pod, v1.ResourceMemory)
+	if cpuRequests > cpuAvailable || memRequests > memAvailable {
+		klog.V(2).InfoS("Resize is not feasible as request exceeds available node resources", "Pod", pod.Name)
+		return false, nil, v1.PodResizeStatusInfeasible
+	}
+
+	// Treat the existing pod needing resize as a new pod with desired resources seeking admit.
+	// If desired resources don't fit, pod continues to run with currently allocated resources.
+	activePods := kl.GetActivePods()
+	for _, p := range activePods {
+		if p.UID != pod.UID {
+			otherActivePods = append(otherActivePods, p)
+		}
+	}
+
+	if ok, failReason, failMessage := kl.canAdmitPod(otherActivePods, pod); !ok {
```

Comment:
I don't understand why we'd preempt this check as part of the normal kubelet sync loop. Can we not update the spec and let it run? That appears to be what will happen anyways if the request exceeds available resources. 

Edit: Looks like @Random-Liu had a similar comment: https://github.com/kubernetes/kubernetes/pull/102884/files/9d0831a0d5d6ecbb24689d4ec6a3763a12e22b45..27724556f8489cb8de82bd324ebe858b510b3abb#r663195259